### PR TITLE
Introduce PoseGraphModel and use it in 2D.

### PR DESCRIPTION
**Pull request by pifon2a**
[PoseGraphData RFC](https://github.com/googlecartographer/rfcs/blob/master/text/0022-pose-graph-data.md) is submitted.

### CODE REVIEW

File: cartographer/mapping/pose_graph_model.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  // IDs of the nodes that were inserted into this map together with
+  // constraints for them. They are not to be matched again when this submap
+  // becomes 'finished'.
+  std::set<NodeId> node_ids;
+};
+
+struct PoseGraphModel {
+  // Submaps get assigned an ID and state as soon as they are seen, even
+  // before they take part in the background computations.
+  MapById<SubmapId, InternalSubmapData> submap_data_;
```
Comment (gaschler) [inline cartographer/mapping/pose_graph_model.h]: style: no trailing underscore in structs

File: cartographer/mapping/pose_graph_model.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  // Set of all initial trajectory poses.
+  std::map<int, PoseGraph::InitialTrajectoryPose> initial_trajectory_poses_;
+
+  std::vector<PoseGraphInterface::Constraint> constraints_;
+};
+
+}  // namespace mapping
+}  // namespace cartographer
+
+#endif  // CARTOGRAPHER_MAPPING_INTERNAL_2D_POSE_GRAPH_2D_H_
```
Comment (gaschler) [inline cartographer/mapping/pose_graph_model.h]: update comment

Comment (pifon2a) [reply cartographer/mapping/pose_graph_model.h]: which one?

Comment (pifon2a) [reply cartographer/mapping/pose_graph_model.h]: done, thanks!

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Pull request merged**